templates/ts-react-apollo-client/README.md

```diff
@@ -35,15 +19,41 @@ yarn install
 ```
 
+## Dependencies And Tools:
+
+The following tools and technologies have been used to create this template:
+
+- [GraphQL](https://graphql.org/): GraphQL is an open-source data query and manipulation language for APIs which was publically released by Facebook in 2015.
+
+- [Apollo Client](https://www.apollographql.com/docs/react/): Apollo Client provides a neat abstraction layer and an interface to interact with the GraphQL server. It allows easier building of UI components that fetch data via GraphQL.
+
+- [Material UI](https://material-ui.com/): Material-UI is an open-source project that features React components that implement Google's Material Design. It has been used for styling this sample React template.
+
+## Project structure
+
+The project contains the following set of source files.
+
+- The `.graphlrc.yml` file defining the configuration like the path to business model declaration, how to perform code generation from the GraphQL types to Typescript types etc. The configuration file is defined using the GraphQL project using [`graphql-config`](https://graphql-config.com/introduction).
+- A `public` folder which contains static files.
+- A `src` folder which has:
+  - A `components` folder defining react components for each type of business model.
+  - A `graphql` folder which contains a generated `graphback.graphql` using the [generate command](#development-and-adjustment-to-schema-changes) and a custom `getDraftNotes.graphql`. You can add additional custom GraphQL files according to your needs.
+  - A `generated-types.tsx` file, which is also generated as indicated by its name. See [Generating types](#development-and-adjustment-to-schema-changes)
+  - The rest of the files are to initialize a standard ReactJs Progressive Web App.
+
+> NOTE: All the files can be edited according to your needs except for those that are generated (no need to edit them as they'll be re-generated anyway).
+
 ## Development and adjustment to schema changes
 
-Client is designed to work only with the sample schema.
-When changing graphback schema you will need to recompile all the changes.
+The client is designed to work only with the sample schema.
+When changing the graphback schema you will need to recompile all the changes.
 
 - Generate React components
 
 ```
 yarn generate
 ```
 
-> NOTE: You have to run the above commands on each modification of your business model in the server side. This ensure that the client is kept in sync with changes on the server side. 
\ No newline at end of file
+Running `yarn generate` executes a script which relies on two things, [Graphback CLI](https://graphback.dev/docs/cli/graphback-cli) and [Codegen](https://graphql-code-generator.com/). The Graphback CLI is used to run the `graphback generate` command which executes the generation process that generates the GraphQL server based on your data model. And codegen is used in the `graphql codegen` command which generates code from the GraphQL schema.
```

Comment:
I'd like to see this paragraph formatted a little better, so have a separate paragraph mentioning Graphback CLI and GraphQL Code Generator. Even better would be if they were under two separate headings (GraphQL Document Generation and Code Generation)

Also, a sentence should not begin with "And"

Reply:
I need to confirm if I've understood this correctly. So ```graphback generate``` creates the schema in the file "graphback.graphql" and then codegen (```graphql codegen```) uses this created schema to further generate the "generated-types.tsx" file. Right?

Reply:
Right. And this is actually really hard for people to understand thus we need it documented - idealy some photo or something.

Reply:
Thanks for clearing it out. Now that I have a good idea I'll make sure others will be able to understand too :)